### ISSUE TITLE
Missing `updateInternalUsername` call in `initiateCustomAuthRequest`

### ISSUE THREAD

**Issue by dbathily** [aws-amplify/aws-sdk-android#657]
**Describe the bug**
When initiating a custom auth without `clientSecret` I have this error:

```
W/System.err: com.amazonaws.mobileconnectors.cognitoidentityprovider.exceptions.CognitoParameterInvalidException: user ID cannot be null
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.util.CognitoSecretHash.getSecretHash(CognitoSecretHash.java:45)
W/System.err:     at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.initiateCustomAuthRequest(CognitoUser.java:2760)
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.startWithCustomAuth(CognitoUser.java:2383)
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.initiateUserAuthentication(CognitoUser.java:775)
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.continuations.AuthenticationContinuation$1.run(AuthenticationContinuation.java:100)
        at java.lang.Thread.run(Thread.java:764)
```

**To Reproduce**

```
val alias = "..."
userPoool.getUser(alias).getSessionInBackground(object: AuthenticationHandler {
            override fun onSuccess(userSession: CognitoUserSession?, newDevice: CognitoDevice?) {
            }

            override fun onFailure(exception: Exception?) {
                exception?.printStackTrace()
            }

            override fun getAuthenticationDetails(authenticationContinuation: AuthenticationContinuation?, userId: String?) {
                val details = AuthenticationDetails(userId, mapOf(
                        CognitoServiceConstants.AUTH_PARAM_USERNAME to userId
                ), null)
                authenticationContinuation?.setAuthenticationDetails(details)
                authenticationContinuation?.continueTask()
            }

            override fun authenticationChallenge(continuation: ChallengeContinuation?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun getMFACode(continuation: MultiFactorAuthenticationContinuation?) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        })
```

**Which AWS service(s) are affected?**

Cognito

**Expected behavior**

Should not raised and exception, internalUsername must be updated with the parameters

**Environment(please complete the following information):**
 - SDK Version: 2.9

**Comment by palpatim** [comment]
Thanks for reporting the issue @dbathily. We will investigate and update this thread when we have more info or questions.

**Comment by anees17861** [comment]
+1

**Comment by minbi** [comment]
Hi @dbathily @anees17861 ,

We have released a fix for this issue in version 2.11.1. Please see if this fixes the issue for you.
